### ISSUE TITLE
(📚) docs for `PLW1641` `known problems` doesn't make sense

### ISSUE THREAD

**Issue by KotlinIsland** [astral-sh/ruff#14883]
```
# Known problems
Does not check for `__hash__` implementations in superclasses.
```

`__hash__` defined in a superclass doesn't affect the semantics:
```py
class A:
    def __hash__(self):
        return 1

class B(A):
   def __eq__(self, other):
       return False
B.__hash__ # None
```

**Comment by zanieb** [comment]
The rule is

> Checks for classes that implement __eq__ but not __hash__.


So.. if you have this:

```python
class A:
    def __hash__(self):
        return 1

class B(A):
   def __eq__(self, other):
       return False
```

B implements __hash__ via A but is not detected by Ruff.

What doesn't make sense about this?

**Comment by KotlinIsland** [comment]
> What doesn't make sense about this?

as in my OP:
```py
print(B.__hash__) # None
```

**Comment by KotlinIsland** [comment]
> A class that overrides [**eq**()](https://docs.python.org/3/reference/datamodel.html#object.__eq__) and does not define [**hash**()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) will have its [**hash**()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) implicitly set to None.

source: https://docs.python.org/3/reference/datamodel.html#object.__hash__

**Comment by charliermarsh** [comment]
Wow, so it's _just_ when `__eq__` is defined?

**Comment by zanieb** [comment]
@KotlinIsland sorry your OP doesn't have inheritance

Interesting find though!
